src/core/shape_inference/include/experimental_detectron_prior_grid_generator_shape_inference.hpp

```diff
@@ -41,22 +42,21 @@ void shape_infer(const ExperimentalDetectronPriorGridGenerator* op,
 
     if (im_data_rank_static) {
         NODE_VALIDATION_CHECK(op, im_data_shape.size() == 4, "Im_data rank must be equal to 4.");
-    }
 
-    if (featmap_rank_static && im_data_rank_static) {
-        const auto& num_batches_featmap = featmap_shape[0];
-        const auto& num_batches_im_data = im_data_shape[0];
+        if (featmap_rank_static) {
+            const auto num_batches = featmap_shape[0] & im_data_shape[0];
 
-        NODE_VALIDATION_CHECK(op,
-                              num_batches_featmap.compatible(num_batches_im_data),
-                              "The first dimension of both 'feature_map' and 'im_data' must match. "
-                              "Feature_map: ",
-                              num_batches_featmap,
-                              "; Im_data: ",
-                              num_batches_im_data);
+            NODE_VALIDATION_CHECK(op,
+                                  num_batches.compatible(1),
```

Comment:
This check is aligned with spec, but can break backward compatibility.
(Previously possible equality to 1 wasn't validated).

Reply:
Yes it looks like current tests have got different assumption and check against dimension `1` is too strict. 
Validation updated.